atoma-state/src/state_manager.rs

```diff
@@ -0,0 +1,1320 @@
+    ///
+    /// This method deletes the entry from the `stacks` table that matches the provided `stack_small_id`.
+    ///
+    /// # Arguments
+    ///
+    /// * `stack_small_id` - The unique small identifier of the stack to remove.
+    ///
+    /// # Returns
+    ///
+    /// - `Result<()>`: A result indicating success (Ok(())) or failure (Err(StateManagerError)).
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if:
+    /// - The database query fails to execute.
+    ///
+    /// # Example
+    ///
+    /// ```rust,no_run
+    /// use your_crate::StateManager;
+    ///
+    /// async fn remove_stack(state_manager: &StateManager, stack_small_id: i64) -> Result<(), StateManagerError> {
+    ///     state_manager.remove_stack(stack_small_id).await
+    /// }
+    /// ```
+    #[tracing::instrument(
+        level = "trace",
+        skip_all,
+        fields(stack_small_id = %stack_small_id)
+    )]
+    pub async fn remove_stack(&self, stack_small_id: i64) -> Result<()> {
+        sqlx::query("DELETE FROM stacks WHERE stack_small_id = ?")
+            .bind(stack_small_id)
+            .execute(&self.db)
+            .await?;
+        Ok(())
+    }
+
+    /// Retrieves the stack settlement ticket for a given stack.
+    ///
+    /// This method fetches the settlement ticket details from the `stack_settlement_tickets` table
+    /// based on the provided `stack_small_id`.
+    ///
+    /// # Arguments
+    ///
+    /// * `stack_small_id` - The unique small identifier of the stack whose settlement ticket is to be retrieved.
+    ///
+    /// # Returns
+    ///
+    /// - `Result<StackSettlementTicket>`: A result containing either:
+    ///   - `Ok(StackSettlementTicket)`: A `StackSettlementTicket` object representing the settlement ticket.
+    ///   - `Err(StateManagerError)`: An error if the database query fails or if there's an issue parsing the results.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if:
+    /// - The database query fails to execute.
+    /// - There's an issue converting the database row into a `StackSettlementTicket` object.
+    /// - No settlement ticket is found for the given `stack_small_id`.
+    ///
+    /// # Example
+    ///
+    /// ```rust,no_run
+    /// use your_crate::{StateManager, StackSettlementTicket};
+    ///
+    /// async fn get_settlement_ticket(state_manager: &StateManager, stack_small_id: i64) -> Result<StackSettlementTicket, StateManagerError> {
+    ///     state_manager.get_stack_settlement_ticket(stack_small_id).await
+    /// }
+    /// ```
+    #[tracing::instrument(
+        level = "trace",
+        skip_all,
+        fields(stack_small_id = %stack_small_id)
+    )]
+    pub async fn get_stack_settlement_ticket(
+        &self,
+        stack_small_id: i64,
+    ) -> Result<StackSettlementTicket> {
+        let stack_try_settle =
+            sqlx::query("SELECT * FROM stack_settlement_tickets WHERE stack_small_id = ?")
```

Comment:
I don't see the `stack_settlement_tickets` table created.

Reply:
That's a big one, thanks for noticing !